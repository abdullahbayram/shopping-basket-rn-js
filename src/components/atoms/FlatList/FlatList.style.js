import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  flatList: {
    marginTop: 7,
  },
});
