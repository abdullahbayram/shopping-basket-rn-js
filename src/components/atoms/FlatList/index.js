import { FlatList as RNFlatList } from 'react-native';
import styles from './FlatList.style';

const FlatList = ({ ...rest }) => {
  return (
    <RNFlatList
      style={styles.flatList}
      initialNumToRender={10}
      maxToRenderPerBatch={5}
      windowSize={10}
      removeClippedSubviews
      {...rest}
    />
  );
};

export default FlatList;
