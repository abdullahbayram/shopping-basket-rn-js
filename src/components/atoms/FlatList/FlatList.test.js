import React from 'react';
import { render, screen } from '@testing-library/react-native';
import FlatList from '.';
import styles from './FlatList.style';

jest.mock('./FlatList.style', () => ({
  flatList: {
    backgroundColor: 'white',
  },
}));

describe('FlatList Component', () => {
  it('should render correctly with default props', () => {
    render(<FlatList testID="custom-flatlist" data={[]} renderItem={() => null} />);

    const flatList = screen.getByTestId('custom-flatlist');

    expect(flatList).toBeTruthy();
    expect(flatList.props.style).toEqual(styles.flatList);
    expect(flatList.props.initialNumToRender).toBe(10);
    expect(flatList.props.maxToRenderPerBatch).toBe(5);
    expect(flatList.props.windowSize).toBe(10);
    expect(flatList.props.removeClippedSubviews).toBe(true);
  });

  it('should pass additional props to the FlatList', () => {
    const mockData = [{ id: '1', title: 'Test Item' }];
    const mockRenderItem = jest.fn();

    render(<FlatList testID="custom-flatlist" data={mockData} renderItem={mockRenderItem} horizontal />);

    const flatList = screen.getByTestId('custom-flatlist');

    expect(flatList.props.data).toEqual(mockData);
    expect(flatList.props.horizontal).toBe(true);
  });
});
