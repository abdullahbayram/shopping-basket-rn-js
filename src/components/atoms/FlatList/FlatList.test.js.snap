// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlatList Component renders correctly with default props (snapshot) 1`] = `
<RCTScrollView
  data={[]}
  getItem={[Function]}
  getItemCount={[Function]}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={5}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={true}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  stickyHeaderIndices={[]}
  style={
    {
      "backgroundColor": "white",
    }
  }
  testID="custom-flatlist"
  viewabilityConfigCallbackPairs={[]}
  windowSize={10}
>
  <View />
</RCTScrollView>
`;
